### ISSUE TITLE
Strange component loading behavior/errors ("The requested module does not provide an export")

### ISSUE THREAD

**Issue by mattdsteele** [ionic-team/stencil#528]
**Stencil version:**

```
 @stencil/core@0.5.0
```

**I'm submitting a:**

- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or https://stencil-worldwide.slack.com

**Current behavior:**

This is a reduction of a larger app and is as far as I could get while still reproducing the error - let me know if any part doesn't make sense.

So I have a few components:

* `<x-foo>`, which is just standalone
* `<x-bar>` and `<x-baz>` which can be nested inside each other. In my original case they had a circular dependency but I can reproduce with an acyclic graph
* Some of the components have `shadow: true`, others `shadow: false` due to some weirdness in how the shadycss polyfills behave

Each component also depends on some shared util functions, under `src/utils`.

When I load them in this particular configuration, the components fail to load when running `npm run dev`. In Chrome there are a few different errors:

```
SyntaxError: The requested module does not provide an export named 'NavLink'
ReferenceError: utilFn is not defined
```

In polyfilled browsers (IE, Edge, Firefox) the error looks a little different:

```
TypeError: cmpConstructor is undefined[Learn More]
ReferenceError: __chunk1_js is not defined
```

**Steps to reproduce:**

1. Clone and install https://github.com/mattdsteele/stencil-missing-chunks
2. `npm run dev`, you should see the errors in the console

**Other information:**

Continuation from this Slack thread: https://stencil-worldwide.slack.com/messages/C944RNLRX/convo/C944RNLRX-1518478340.000208/

If you'd prefer to chat there, let me know!

**Comment by mattdsteele** [comment]
Looks like I can get around the `utilFn is not defined` issue - it looks like re-exporting a common function causes problems, i.e.

```
// foo.ts uses re-exported utilFn
import { utilFn } from '../../utils';

// bar.ts uses the original utilFn
import { utilFn } from '../../../utils/util-fn';
```

Always using the re-exported function seems to be a usable workaround.

Still unclear why the "Does not provide an export" issue is cropping up.

**Comment by mattdsteele** [comment]
I think the `The requested module does not provide an export named 'NavLink'` is also the result of one component exporting a constant, and me using it in another component:

https://github.com/mattdsteele/stencil-missing-chunks/blob/master/src/components/navigation/nav-link/bar.tsx#L4

Moving this export to a separate file seems to be resolve it.

**Comment by adamdbradley** [comment]
@mattdsteele Would you be able to test with `0.5.1`?

**Comment by mattdsteele** [comment]
Just gave 0.5.2 a shot and I'm still seeing these issues. Updated the repo and `npm start` should still show the error in Chrome & Firefox. (I'm building on Windows 10 & Node 6, if that makes a difference)

**Comment by jthoms1** [comment]
These appear to both be issues with Rollup.  I am going to create some examples that do not use Stencil to try to isolate the issues.

**Comment by jthoms1** [comment]
It appears this was fixed in rollup.  This comment resolves the issue 3d744121.
